programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -17,15 +17,13 @@ use crate::{
     output_token: Pubkey,
     input_amount: u64,
     output_amount: u64,
-    destination_chain_id: u64, // TODO: we can remove some of these instructions props
-    exclusive_relayer: Pubkey,
-    quote_timestamp: u32,
-    fill_deadline: u32,
-    exclusivity_deadline: u32,
-    message: Vec<u8>
+    destination_chain_id: u64,
 )]
 pub struct DepositV3<'info> {
-    #[account(mut)]
+    #[account(
+        mut,
+        constraint = signer.key() == depositor @ SvmError::DepositorMustBeSigner
```

Comment:
We don't have this restriction in EVM where sender can provide any depositor address

Reply:
you're right, we dont really care about this as long as the authority is the depositor. previously we checked the authority is the signer, but now we check it is the depositor, which is functionally different. the goal I was trying to achieve is that deposit will work for a multi-sig & correct ATA support, where the depositor is the owner of the ATA. I think how it's structured, without this constraint, will support multisigs where the authority _is not the signer_.

Reply:
yes, we don't really care that depositor is signer. and in case of multisig the owner will be the token program. we don't need to replicate any authorsation checks here as they will be performed in CPI transfer